### PR TITLE
feat(CosmosPackagesActions): [1 of 3] select package version actions/events/types

### PR DESCRIPTION
**Pull request by weblancaster**
_ps: to test out the whole feat you will need #2397 and #2396_

Commit including only the actions, event types and unit test for them.

Closes DCOS_OSS-1560 DCOS_OSS-1559 DCOS_OSS-1558

**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [x] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

### CODE REVIEW

**Comment by MatApple** [comment]
@weblancaster I think this doesn't depend on those 2 PRs, those 2 PRs depend on this right?

**Comment by MatApple** [comment]
Alfred retest this please

File: src/js/events/CosmosPackagesActions.js [hunk 1]
```diff
@@ -149,6 +150,35 @@ const CosmosPackagesActions = {
...
+    // necessary for the backend
+    // even though it's not in use
+    const includePackageVersions = false;
+
+    RequestUtil.json({
+      contentType: getContentType("list-versions", "request", "v1"),
+      headers: { Accept: getContentType("list-versions", "response", "v1") },
+      method: "POST",
+      url: `${Config.rootUrl}${Config.cosmosAPIPrefix}/list-versions`,
+      data: JSON.stringify({ packageName, includePackageVersions }),
```
Comment (MatApple) [inline src/js/events/CosmosPackagesActions.js]: What happens if `includePackageVersions` is not included in the request?

File: src/js/events/CosmosPackagesActions.js [hunk 2]
```diff
@@ -290,6 +320,7 @@ if (Config.useFixtures) {
   const packagesListFixture = require("../../../tests/_fixtures/cosmos/packages-list.json");
   const packagesSearchFixture = require("../../../tests/_fixtures/cosmos/packages-search.json");
   const packagesRepositoriesFixture = require("../../../tests/_fixtures/cosmos/packages-repositories.json");
+  const packageListVersionsFixture = require("../../../tests/_fixtures/cosmos/package-list-versions.json");
```
Comment (MatApple) [inline src/js/events/CosmosPackagesActions.js]: We should include this file in this PR.

File: src/js/events/__tests__/CosmosPackagesActions-test.js [hunk 3]
```diff
@@ -220,6 +220,126 @@ describe("CosmosPackagesActions", function() {
...
+        AppDispatcher.unregister(id);
+        expect(action.type).toEqual(
+          ActionTypes.REQUEST_COSMOS_PACKAGE_LIST_VERSIONS_ERROR
+        );
+      });
+
+      this.configuration.error({
+        responseJSON: {
+          results: {
+            "1.0.2": "0"
```
Comment (MatApple) [inline src/js/events/__tests__/CosmosPackagesActions-test.js]: Perhaps we should make this response look like an "error" message.

File: src/js/events/__tests__/CosmosPackagesActions-test.js [hunk 4]
```diff
@@ -220,6 +220,126 @@ describe("CosmosPackagesActions", function() {
...
+      var id = AppDispatcher.register(function(payload) {
+        var action = payload.action;
+        AppDispatcher.unregister(id);
+        expect(action.data).toEqual("An error has occurred.");
+      });
+
+      this.configuration.error({
+        responseJSON: {
+          results: {
+            "1.0.2": "0"
```
Comment (MatApple) [inline src/js/events/__tests__/CosmosPackagesActions-test.js]: Perhaps we should make this response look like an "error" message.

File: src/js/events/__tests__/CosmosPackagesActions-test.js [hunk 5]
```diff
@@ -220,6 +220,126 @@ describe("CosmosPackagesActions", function() {
...
+          responseJSON: {
+            results: {
+              "1.0.2": "0"
+            }
+          }
+        });
+      });
+
+      this.configuration.error({
+        responseJSON: {
```
Comment (MatApple) [inline src/js/events/__tests__/CosmosPackagesActions-test.js]: Perhaps we should make this response look like an "error" message.

File: src/js/events/__tests__/CosmosPackagesActions-test.js (hunk 4 above)
Comment (MatApple) [reply src/js/events/__tests__/CosmosPackagesActions-test.js]: We should also make sure that we test for the result of `RequestUtil.getErrorFromXHR(xhr)`.

File: src/js/events/CosmosPackagesActions.js (hunk 1 above)
Comment (weblancaster) [reply src/js/events/CosmosPackagesActions.js]: the server denies the request and returns an error.

File: src/js/events/__tests__/CosmosPackagesActions-test.js (hunk 3 above)
Comment (weblancaster) [reply src/js/events/__tests__/CosmosPackagesActions-test.js]: Makes sense. I was following what we have in place for other unit tests.

**Comment by weblancaster** [comment]
@MatApple, when I say, depends on is that, it's ok to merge since won't break anything but to actually see working in the UI because we should be testing all 3 PRs together.

File: src/js/events/CosmosPackagesActions.js [hunk 6]
```diff
@@ -127,9 +129,8 @@ const CosmosPackagesActions = {
       data: JSON.stringify({ packageName, packageVersion }),
       success(response) {
         const cosmosPackage = response.package;
-        if (!cosmosPackage.currentVersion && cosmosPackage.version) {
-          cosmosPackage.currentVersion = cosmosPackage.version;
-        }
+
+        cosmosPackage.currentVersion = cosmosPackage.version;
```
[the remark this answers is not in the document]
Comment (weblancaster) [reply src/js/events/CosmosPackagesActions.js]: I will confirm today that won't break anything and remove.

File: src/js/events/CosmosPackagesActions.js (hunk 2 above)
Comment (weblancaster) [reply src/js/events/CosmosPackagesActions.js]: Yes I will include the file.

**Comment by MatApple** [comment]
@weblancaster I think in general, it won't get merged if it say's it depends on other PRs if they are not merged. Please only use "depends on" for PRs that cannot merge till the referenced PRs have merged.

**Comment by MatApple** [comment]
alfred please run tests

**Comment by MatApple** [comment]
Alfred retest this plesae

**Comment by MatApple** [comment]
Thank you Alfred

**Pull request merged**